tests/draft4/ref.json

```diff
@@ -174,6 +174,35 @@
         ]
     },
     {
+        "description": "nested ref with other properties",
+        "schema": {
+            "definitions": {
+                "a": {
+                    "type": "number"
+                }
+            },
+            "properties": {
+                "foo": {
+                    "type": "number",
+                    "maximum": 5,
+                    "$ref": "#/definitions/a"
+                }
```

Comment:
Or have I misunderstood the current behaviour for additonal properties alongside a $ref? These tests assume that any additonal properties are ignored/overwitten by those in a $ref, my reading of the json ref spec is that this is correct, but I'm not 100% sure. Either way if someone could explicitly clarify for me here I'll make the appropriate changes